frontend/src/metabase/admin/permissions/selectors/data-permissions/group-sidebar.unit.spec.ts

```diff
@@ -1,20 +1,46 @@
 import type { State } from "metabase-types/store";
 
-import { DataPermissionValue } from "../../types";
+import { DataPermission, DataPermissionValue } from "../../types";
 
 import { state as mockState } from "./data-permissions.unit.spec.fixtures";
 
 import { getGroupsDataPermissionEditor } from ".";
 
-const state = mockState as unknown as State;
+const stateWithoutLegacyValues = mockState as unknown as State;
+
+// adding legacy no self-service in the graph will prevent getGroupsDataPermissionEditor
+// from omitting the view data permission options in the case there's only one option
+const stateWithLegacyValues = {
```

Comment:
`assocIn` from `icepick` might be handy here haha.

Reply:
man i thought it was crazy we didn't have a deep merge method... thank goodness. fixed!